GitCommands/PathUtil.cs

```diff
@@ -124,14 +124,47 @@ public static string NormalizePath([NotNull] this string path)
 
             try
             {
-                return Path.GetFullPath(new Uri(path).LocalPath);
+                return Path.GetFullPath(PathUtil.SolveUriLocalPath(path));
             }
             catch (UriFormatException)
             {
                 return string.Empty;
             }
         }
 
+        [NotNull]
+        public static string SolveUriLocalPath([NotNull] string path, [NotNull] string relativePath = "")
+        {
+            // Fix for on purpose invalid machine name of WSL in Windows 10
+            var wslPathReplaced = false;
+            if (path.StartsWith(@"\\wsl$\"))
+            {
+                wslPathReplaced = true;
```

Comment:
could be shortened and be made more explicit
```suggestion
            bool replaceWslPath = path.StartsWith(@"\\wsl$\");
            if (replaceWslPath)
            {
```